src/eth/types.ts

```diff
@@ -11,3 +15,87 @@ export const EthUint256Struct = definePattern(
   'EthUint256',
   /^0x([1-9a-f][0-9a-f]*|0)$/iu,
 );
+
+/**
+ * Supported Ethereum methods.
+ */
+export enum EthMethod {
+  // General signing methods
+  PersonalSign = 'personal_sign',
+  Sign = 'eth_sign',
+  SignTransaction = 'eth_signTransaction',
+  SignTypedDataV1 = 'eth_signTypedData_v1',
+  SignTypedDataV3 = 'eth_signTypedData_v3',
+  SignTypedDataV4 = 'eth_signTypedData_v4',
+}
+
+/**
+ * Supported Ethereum methods for ERC-4337 (Account Abstraction) accounts.
+ */
+export enum EthErc4337Method {
+  // ERC-4337 methods
+  PrepareUserOperation = 'eth_prepareUserOperation',
+  PatchUserOperation = 'eth_patchUserOperation',
+  SignUserOperation = 'eth_signUserOperation',
+}
+
+/**
+ * Supported Ethereum account types.
+ */
+export enum EthAccountType {
+  Eoa = 'eip155:eoa',
+  Erc4337 = 'eip155:erc4337',
+}
+
+export const EthEoaAccountStruct = object({
```

Comment:
Should this be called `EthEOAccount` instead of `EthEoaAccount`?

Reply:
Good question, I tend to use "whole" word for acronyms, I've seen this naming convention in other contexts/languages.

Also, based on this one, we should use "whole" words: https://google.github.io/styleguide/tsguide.html#camel-case

Couldn't find anything here either: https://github.com/MetaMask/contributor-docs/blob/main/docs/typescript.md

So I would stick with `Eoa` here. What's your opinions on this @danroc @montelaidev?

Reply:
That's ok in my opinion, I'll resolve this comment